libopae/plugins/xfpga/sysfs.c

```diff
@@ -1860,11 +1953,9 @@ fpga_result make_sysfs_group(char *sysfspath, const char *name,
 	return FPGA_OK;
 
 out_free_group:
-	if (group->path)
-		free(group->path);
-	if (group->name)
-		free(group->name);
-	free(group);
+	if (destroy_fpga_object(group)) {
+		FPGA_ERR("Error destroying object");
```

Comment:
Catch error code / propagate it upwards?

Reply:
If we're in here, it was because of an error, I don't want to overwrite the error with another one.